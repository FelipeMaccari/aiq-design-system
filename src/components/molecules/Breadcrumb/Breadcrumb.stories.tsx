import React from 'react'
import { MdHome } from 'react-icons/md'

import { Breadcrumb, BreadcrumbItem } from './Breadcrumb'
import { Link } from '../../atoms/Link'

export default {
  component: Breadcrumb,
  title: 'molecules/Breadcrumb'
}

export const Basic: React.FC = () => (
  <Breadcrumb>
    <BreadcrumbItem>
      <MdHome />
      <Link marginLeft='5px' href='#'>
        início
      </Link>
    </BreadcrumbItem>
    <BreadcrumbItem>
      <Link href='#'>relatório</Link>
    </BreadcrumbItem>
    <BreadcrumbItem>
      <Link href='#'>pedidos</Link>
    </BreadcrumbItem>
  </Breadcrumb>
)
