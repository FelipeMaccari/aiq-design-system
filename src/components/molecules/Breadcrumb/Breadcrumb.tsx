import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

export interface Props {
  children?: any
}

const BreadcrumbStyled = styled.ul`
  list-style: none;
  display: flex;
  flex-direction: row;
`

const BreadcrumbItemStyled = styled.li`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.darkGrey};
  font-weight: 500;

  a {
    color: ${({ theme }) => theme.colors.darkGrey};
  }

  &:last-child {
    color: ${({ theme }) => theme.colors.primary};
    a {
      color: ${({ theme }) => theme.colors.primary};
    }

    &:after {
      content: '';
    }
  }

  &:after {
    margin-right: 8px;
    margin-left: 8px;
    color: ${({ theme }) => theme.colors.darkGrey};
    content: ' / ';
  }
`

export const BreadcrumbItem: React.FC<Props> = ({ children }) => {
  return <BreadcrumbItemStyled>{children}</BreadcrumbItemStyled>
}

BreadcrumbItem.propTypes = {
  children: PropTypes.any
}

export const Breadcrumb: React.FC<Props> = ({ children }) => {
  return <BreadcrumbStyled>{children}</BreadcrumbStyled>
}

Breadcrumb.propTypes = {
  children: PropTypes.any
}
