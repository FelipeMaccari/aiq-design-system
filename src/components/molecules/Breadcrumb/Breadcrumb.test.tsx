import React from 'react'
import renderer from 'react-test-renderer'

import { Breadcrumb } from './Breadcrumb'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Breadcrumb />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
