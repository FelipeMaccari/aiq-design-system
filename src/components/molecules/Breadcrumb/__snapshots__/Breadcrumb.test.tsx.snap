// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<ul
  className="sc-AxirZ iJBhLN"
>
  <li
    className="sc-AxiKw gNSxAH"
  >
    <a
      className="sc-AxheI iBNNrJ"
      href="#"
    >
      início
    </a>
  </li>
  <li
    className="sc-AxiKw gNSxAH"
  >
    <a
      className="sc-AxheI juDcHF"
      href="#"
    >
      relatório
    </a>
  </li>
  <li
    className="sc-AxiKw gNSxAH"
  >
    <a
      className="sc-AxheI juDcHF"
      href="#"
    >
      pedidos
    </a>
  </li>
</ul>
`;
